env0/resource_template.go

```diff
@@ -344,11 +344,17 @@ func templateCreatePayloadFromParameters(prefix string, d *schema.ResourceData)
 	}
 
 	tokenIdKey := "token_id"
+	isAzureDevOpsKey := "is_azure_devops"
+
 	if prefix != "" {
 		tokenIdKey = prefix + "." + tokenIdKey
+		isAzureDevOpsKey = prefix + "." + isAzureDevOpsKey
 	}
+
 	if tokenId, ok := d.GetOk(tokenIdKey); ok {
-		payload.IsGitLab = tokenId != ""
+		if isAzureDevOps := d.Get(isAzureDevOpsKey); !isAzureDevOps.(bool) {
```

Comment:
only set implicit isGitlab if is_azure_devops is false.
This is more backward compatible.

Reply:
Not sure I understand the logic here.

What's the prefix is used for? how come we assert it from the existing schema although it's not static? I'm really not in the details so an explanation would be superb please 🙏

Reply:
Ahh no problems!
A standalone template and an environment with a template inside use the same code and schema.
(It took a bit of time to get it done correctly, but now any changes in the template apply to both use cases).
In some cases, an "if" is required.

Reply:
Would appreciate adding some comment on that as code is (rightfully) quite generic on this case

Reply:
Hopefully, I understood what you meant. I added a comment.

Reply:
0cdf4d42c5c171549e579fad7f7c3f690736faeb